### ISSUE TITLE
Cleanup docu (eliminate oasp)

### ISSUE THREAD

**Issue by hohwille** [devonfw/devon4j#182]
OASP (oasp.io) is a pre-decessor of devonfw and is officially deprecated and replaced by devonfw. The devonfw documentation should avoid talking about or referencing to OASP (except for maybe a historical explanation or some migration guides in `devonfw-ide`).
Hence we should clean our documentation from OASP where easily possible:

```
cd devon4j/documentation
grep oasp *
```

> devonfw-ide-setup.asciidoc:* If you are a member of Capgemini: download https://coconet.capgemini.com/sf/go/projects.apps2_devon/frs.oasp4j_ide[devonfw ide package] or the higher integrated https://coconet.capgemini.com/sf/go/projects.apps2_devon/frs.devon_distribution[devonfw distribution] (for devonfw please find the setup guide within the devon-dist).
guide-feature-toggle.asciidoc:For a pretty minimal working example, also see https://github.com/florianluediger/oasp4j[this fork].
guide-feature-toggle.asciidoc:The following example takes place in the `oasp-sample-core` project, so the necessary dependencies have to be added to the according `pom.xml` file. Required are the main Togglz project including Spring support, the Togglz console to graphically change the feature state and the Spring security package to handle authentication for the Togglz console.
guide-feature-toggle.asciidoc:For small features, a simple query of the toggle state is often enough to achieve the desired functionality. To illustrate this, a simple example follows, which implements a toggle to limit the page size returned by the staffmanagement. See https://github.com/florianluediger/oasp4j/commit/e55c3c7cfcb42efe4f279dc673cced730abd580a[here] for further details.
guide-feature-toggle.asciidoc:togglz.feature-enums=io.oasp.gastronomy.restaurant.staffmanagement.featuremanager.StaffmanagementFeatures
guide-feature-toggle.asciidoc:The implementation is pretty straight forward so use this as a reference. Compare https://github.com/florianluediger/oasp4j/commit/8f43f788d3a4a61b6b003c22a8b6c0d3f2254d14[this] for further details.
guide-jpa.asciidoc:Assume an entity `A` contains a collection of `B` entities. Should there be a locking conflict if one user modifies an instance of `A` while another user in parallel modifies an instance of `B` that is contained in the other instance? To address this , take a look at https://oasp.github.io/oasp4j_content/2.4.0/maven/apidocs/io/oasp/module/jpa/dataaccess/api/GenericDao.html#forceIncrementModificationCounter(E)[GenericDao.forceIncrementModificationCounter].
guide-rest.asciidoc:The protocol to communicate these validation errors to the client is worked on at https://github.com/oasp/oasp4j/issues/218
guide-security.asciidoc:|http://repo1.maven.org/maven2/io/oasp/java/templates/[application template] and link:guide-configuration.asciidoc#security[sensitive configuration]
guide-testing.asciidoc://We could replace this with a test example from io.oasp.application.mtsj.dishmanagement.logic.impl
guide-testing.asciidoc:The mocked objects do not provide any functionality at the time being. To define what happens on a method call on a mocked dependency in the CUT one can use `when(_condition_).thenReturn(_result_)`. In this case we want to test `findStaffMember(Long id)` in the https://github.com/oasp/oasp4j/blob/master/samples/core/src/main/java/io/oasp/gastronomy/restaurant/staffmanagement/logic/impl/StaffmanagementImpl.java[StaffmanagementImpl].
guide-testing.asciidoc:devon4j provides a simple test infrastructure to aid with the implementation of subsystem tests. It becomes available by simply subclassing link:https://github.com/oasp/oasp4j/blob/master/samples/core/src/test/java/io/oasp/gastronomy/restaurant/general/common/base/AbstractRestServiceTest.java[AbstractRestServiceTest.java].
guide-testing.asciidoc:As an example let us go to the class https://github.com/oasp/oasp4j/blob/master/samples/core/src/main/java/io/oasp/gastronomy/restaurant/tablemanagement/logic/api/Tablemanagement.java[Tablemanagement]. When testing the method _deleteTable()_ there are several scenarios that can happen and thus should be covered by tests.
guide-testing.asciidoc:* One needs permission to delete a table https://github.com/oasp/oasp4j/blob/master/samples/core/src/main/java/io/oasp/gastronomy/restaurant/general/common/api/constants/PermissionConstants.java[PermissionConstants.DELETE_TABLE]
guide-testing.asciidoc:* The table to delete is required to be https://github.com/oasp/oasp4j/blob/master/samples/core/src/main/java/io/oasp/gastronomy/restaurant/tablemanagement/common/api/datatype/TableState.java[TableState.FREE]
guide-testing.asciidoc:Sometimes it can become handy to provide other or differently configured bean implementations via CDI than those available in production. For example, when creating beans using `@Bean`-annotated methods they are usually configured within those methods. https://github.com/oasp/oasp4j/blob/master/samples/core/src/main/java/io/oasp/gastronomy/restaurant/general/service/impl/config/WebSecurityBeansConfig.java[WebSecurityBeansConfig] shows an example of such methods.
guide-validation.asciidoc://io.oasp.application.mtsj.bookingmanagement.service.rest
guide-validation.asciidoc://io.oasp.application.mtsj.bookingmanagement.logic.api.to
guide-validation.asciidoc://io.oasp.application.mtsj.bookingmanagement.logic.api.to
guide-validation.asciidoc://io.oasp.application.mtsj.bookingmanagement.logic.impl

OASP is dead. 
* `io.oasp` --> `com.devonfw`
* `oasp4j` --> `devon4j`
* `oasp` --> `devonfw`

For `devonfw-ide-setup.asciidoc` delete the entire document as this is replaced by `devonfw-ide` that has its own documentation and nothing in left to say about this in `devon4j`.
The `guide-feature-toggle.asciidoc` might be somewhat more tricky. Maybe this should be kept out of scope of this issue by creating a new issue only for that purpose (with that issue we could create a new sample in the planned devonfw sample organisation on github - aks malte on that - and give a minimal demo for a feature toggle there in an isolated context or even easier link an existing example we found on the web if that exists).
For the rest IMHO things should be easy to fix. IMHO most of the commonts starting with // are from someone who has been to shy to simply remove obsolete content from the documentation. Just strip this stuff out and we are all happy.

**Comment by dhelmke** [comment]
These links:
 florianluediger/oasp4j@e55c3c7 ,  
florianluediger/oasp4j@8f43f78 ,  
https://github.com/florianluediger/oasp4j[this fork] 
 raise an 'Page not found' 404 error for me. 
As well as most of the other links.
Any hint how I can fix this? I would be nice!

**Comment by dhelmke** [comment]
I got it now. It is just the output from grep. Sorry, my bad.

**Comment by dhelmke** [comment]
Sorry for asking again. In the guide-jpa.asciidoc I found a link with:

 https://oasp.github.io/oasp4j_content/2.4.0/maven/apidocs/io/oasp/module/jpa/dataaccess/api/GenericDao.html#forceIncrementModificationCounter(E)

How should I replace that one?

**Comment by hohwille** [comment]
* https://github.com/devonfw/devon4j/blob/develop/modules/jpa-basic/src/main/java/com/devonfw/module/jpa/dataaccess/api/feature/FeatureForceIncrementModificationCounter.java
* https://github.com/devonfw/devon4j/blob/develop/modules/jpa-dao/src/main/java/com/devonfw/module/jpa/dataaccess/base/AbstractGenericDao.java#L115
* https://github.com/devonfw/devon4j/blob/develop/modules/jpa-spring-data/src/main/java/com/devonfw/module/jpa/dataaccess/api/data/GenericRepository.java#L31
* https://github.com/devonfw/devon4j/blob/develop/modules/jpa-spring-data/src/main/java/com/devonfw/module/jpa/dataaccess/impl/data/GenericRepositoryImpl.java#L53

**Comment by dhelmke** [comment]
Thank you for your fast reply. I assume it is the first link I should use?

In the guide-rest.asciidoc there is this link:

 https://github.com/oasp/oasp4j/issues/218

How can I replace that one?

**Comment by dhelmke** [comment]
In guide-security.asciidoc I do not know how to replace this link:

http://repo1.maven.org/maven2/io/oasp/java/templates/

Is there a more general rule how I can replace the link or should I research all the topics in devon4j?

**Comment by dhelmke** [comment]
In guide-testing.asciidoc I found these links:

https://github.com/oasp/oasp4j/blob/master/samples/core/src/main/java/io/oasp/gastronomy/restaurant/general/service/impl/config/WebSecurityBeansConfig.java

https://github.com/oasp/oasp4j/blob/master/samples/core/src/main/java/io/oasp/gastronomy/restaurant/staffmanagement/logic/impl/StaffmanagementImpl.java

https://github.com/oasp/oasp4j/blob/master/samples/core/src/test/java/io/oasp/gastronomy/restaurant/general/common/base/AbstractRestServiceTest.java

https://github.com/oasp/oasp4j/blob/master/samples/core/src/main/java/io/oasp/gastronomy/restaurant/tablemanagement/logic/api/Tablemanagement.java

https://github.com/oasp/oasp4j/blob/master/samples/core/src/main/java/io/oasp/gastronomy/restaurant/general/common/api/constants/PermissionConstants.java

https://github.com/oasp/oasp4j/blob/master/samples/core/src/main/java/io/oasp/gastronomy/restaurant/tablemanagement/common/api/datatype/TableState.java

https://github.com/oasp/oasp4j/blob/master/samples/core/src/main/java/io/oasp/gastronomy/restaurant/general/service/impl/config/WebSecurityBeansConfig.java

In the devonfw/devon4j repo there seems to be no equivalent repo. Has there been some renaming?

**Comment by thoptr-cap** [comment]
MyThaiStar sample implementation was moved to https://github.com/devonfw/my-thai-star/

Thus, for example,  we need to replace:

https://github.com/oasp/oasp4j/blob/master/samples/core/src/main/java/io/oasp/gastronomy/restaurant/general/service/impl/config/WebSecurityBeansConfig.java

by

https://github.com/devonfw/my-thai-star/blob/master/java/mtsj/core/src/main/java/com/devonfw/application/mtsj/general/service/impl/config/WebSecurityBeansConfig.java

**Comment by dhelmke** [comment]
Thank you for your reply. With that information I could replace some of the links. For these links there seems to be no match in My-Thai-Star:

https://github.com/oasp/oasp4j/blob/master/samples/core/src/main/java/io/oasp/gastronomy/restaurant/staffmanagement/logic/impl/StaffmanagementImpl.java

https://github.com/oasp/oasp4j/blob/master/samples/core/src/main/java/io/oasp/gastronomy/restaurant/tablemanagement/logic/api/Tablemanagement.java

https://github.com/oasp/oasp4j/blob/master/samples/core/src/main/java/io/oasp/gastronomy/restaurant/tablemanagement/common/api/datatype/TableState.java

If someone could give me a hint how to replace these links?

**Comment by dhelmke** [comment]
Hi,

replacing some oft the links seems difficult for me without any help.

In guide-testing.asciidoc.

I would suggest replacing:

https://github.com/oasp/oasp4j/blob/master/samples/core/src/main/java/io/oasp/gastronomy/restaurant/tablemanagement/logic/api/Tablemanagement.java

with:

https://github.com/devonfw/my-thai-star/blob/develop/java/mtsj/api/src/main/java/com/devonfw/application/mtsj/bookingmanagement/logic/api/Bookingmanagement.java

Is that correct?
The Permission Constant Part seems to have changed a lot. So replacing this link:

https://github.com/oasp/oasp4j/blob/master/samples/core/src/main/java/io/oasp/gastronomy/restaurant/general/common/api/constants/PermissionConstants.java

with:

https://github.com/devonfw/my-thai-star/blob/develop/java/mtsj/core/src/main/java/com/devonfw/application/mtsj/general/common/api/constants/PermissionConstants.java

only seems to reference the same file, but when I look into the source code, it is not. I can not find the Permission Constants in the MTSJ repo. Where have they been moved?

Since I do not find the Permission Constants, I am not able to replace these link:

https://github.com/oasp/oasp4j/blob/master/samples/core/src/main/java/io/oasp/gastronomy/restaurant/tablemanagement/common/api/datatype/TableState.java

The Staffmanagement seems to have changed a lot as well. So this link:

https://github.com/oasp/oasp4j/blob/master/samples/core/src/main/java/io/oasp/gastronomy/restaurant/staffmanagement/logic/impl/StaffmanagementImpl.java

I do find some similarities in:

https://github.com/devonfw/my-thai-star/blob/develop/java/mtsj/core/src/main/java/com/devonfw/application/mtsj/usermanagement/logic/impl/UsermanagementImpl.java

but I am not sure if that is correct?

As Thorsten suggested I should collect all my questions at once and he did not know how to replace the following links as well. I post them here again:

In guide-rest.asciidoc

https://github.com/oasp/oasp4j/issues/218


In guide-security.asciidoc:

http://repo1.maven.org/maven2/io/oasp/java/templates/

How should I replace these links?

**Comment by hohwille** [comment]
There are different aspects in this testing guide:
* Example for using Mockito: Needs to be updated to JUnit5. An example can be found here: https://github.com/devonfw/my-thai-star/blob/1825456aa52711c1a4103d4428e0728d1e0f0bd8/java/mtsj/core/src/test/java/com/devonfw/application/mtsj/clustermanagement/logic/impl/ClustermanagementTest.java - However, I think that the demonstration of Mockito does not have to be related to our example app (MTS) in any way. You can also get instructions/example from the web - e.g.: https://dzone.com/articles/spring-boot-2-with-junit-5-and-mockito-2-for-unit
* Example for subsystem-tests: Here the entire section is titled `TODO` so the current state is a mess already. We could simply give our REST testing recepie here: https://github.com/devonfw/my-thai-star/blob/ac7f2f51d109d8e9d9fdc1aade7f662f58f87024/java/mtsj/core/src/test/java/com/devonfw/application/mtsj/general/common/base/AbstractRestServiceTest.java - However, also I am thinking that this testing guide should stay rather highlevel. We should more go into a direction that when we explain features like REST or JPA that we should include a section on testing. The testing guide might link to all such places. Or is it better to have everything in that testing guide? The latter question is more regarding the community and experts than you @dhelmke - IMHO you should try to adress the first point. For this point, just create a new issue with the current section (as link or via Copy&Paste) and remove the entire TODO section for the moment, to the your PR done.

**Comment by hohwille** [comment]
> In guide-security.asciidoc:
> http://repo1.maven.org/maven2/io/oasp/java/templates/

The new link is:
http://repo1.maven.org/maven2/com/devonfw/java/templates/
Even better would be to link to this:
https://github.com/devonfw/devon4j/blob/develop/documentation/tutorial-newapp.asciidoc

**Comment by hohwille** [comment]
> In guide-rest.asciidoc
oasp/oasp4j#218

just replace with:

The protocol to communicate these validation errors is described in `xlink:rest-exception-handling[REST exception handling]`.

**Comment by hohwille** [comment]
Done with PR #187
